libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -129,16 +129,14 @@ bool IRGeneratorForStatements::visit(Assignment const& _assignment)
 
 void IRGeneratorForStatements::endVisit(BinaryOperation const& _binOp)
 {
-	solUnimplementedAssert(_binOp.getOperator() == Token::Add, "");
 	solUnimplementedAssert(*_binOp.leftExpression().annotation().type == *_binOp.rightExpression().annotation().type, "");
 	if (IntegerType const* type = dynamic_cast<IntegerType const*>(_binOp.annotation().commonType))
 	{
-		solUnimplementedAssert(!type->isSigned(), "");
 		m_code <<
 			"let " <<
 			m_context.variable(_binOp) <<
 			" := " <<
-			m_utils.overflowCheckedUIntAddFunction(type->numBits()) <<
+			m_utils.intOperatorToLLFunctionString(_binOp.getOperator(), *type) <<
```

Comment:
If you simply replace the function it loses the overflow checks.

Reply:
Yes, but it can be implemented. 
I think in Solidity, overflow checking should not be default.

Reply:
Why not? I guess this needs to be discussed anyway if a feature is to be removed like that, since it was recently introduced.
In the issue you opened previously (#6551) you mentioned `In Solidity, overflow checking is specified by safemath`. This is not true. SafeMath is one library that may be used for that but it is not enforced.

Reply:
@leonardoalt Thanks for your reply.
I want to make sure I realize them correctly. In Solidity, arithmetic operations do not imply overflow checking. It is not a feature now according to [Solidity Doc](https://solidity.readthedocs.io/en/v0.5.7/types.html?highlight=overflow#addition-subtraction-and-multiplication). So my understanding is that no reason overflow checking is implemented here now. If in the future, this feature is in specification, we can modify this piece of code.

If I am obsolete with this issue, very sorry about that.
In that case, I would like to implement the overflow-checking one then reopen this PR.